lib/cache/memory-cache-store.js

```diff
@@ -0,0 +1,133 @@
+'use strict'
+
+/**
+ * @typedef {import('../../types/cache-interceptor.d.ts').default.CacheStore} CacheStore
+ * @implements {CacheStore}
+ */
+class MemoryCacheStore {
+  /**
+   * @type {import('../../types/cache-interceptor.d.ts').default.MemoryCacheStoreOpts} opts
+   */
+  #opts = {}
+  #entryCount = 0
+  /**
+   * @type {Map<string, import('../../types/cache-interceptor.d.ts').default.CacheStoreValue[]>}
+   */
+  #data = new Map()
+
+  /**
+   * @param {import('../../types/cache-interceptor.d.ts').default.MemoryCacheStoreOpts | undefined} opts
+   */
+  constructor (opts) {
+    this.#opts = opts ?? {}
+
+    if (!this.#opts.maxEntries) {
+      this.#opts.maxEntries = Infinity
+    }
+
+    if (!this.#opts.maxEntrySize) {
+      this.#opts.maxEntrySize = Infinity
+    }
+  }
+
+  get entryCount () {
+    return this.#entryCount
+  }
+
+  get maxEntries () {
+    return this.#opts.maxEntries
+  }
+
+  get maxEntrySize () {
+    return this.#opts.maxEntrySize
+  }
+
+  /**
+   * @param {import('../../types/dispatcher.d.ts').default.RequestOptions} req
+   * @returns {import('../../types/cache-interceptor.d.ts').default.CacheStoreValue | undefined}
+   */
+  get (req) {
+    const key = this.#makeKey(req)
+    if (!key) {
+      return undefined
+    }
+
+    const values = this.#data.get(key)
+    if (!values) {
+      return
+    }
+
+    let value
+    const now = Date.now()
```

Comment:
```suggestion
    const now = fastTimers.now()
```

Reply:
Maybe use fastTimers? But be careful, fastTimers.now() returns the since since start of the process.